src/Controller/ResetPasswordControllerTrait.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SymfonyCasts\Bundle\ResetPassword\Controller;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+/**
+ * Provides useful methods to a "reset password controller"
+ *
+ * Use of this trait requires a controller to extend
+ * Symfony\Bundle\FrameworkBundle\Controller\AbstractController
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+trait ResetPasswordControllerTrait
+{
+    private function setCanCheckEmailInSession(): void
+    {
+        $this->getSession()->set('ResetPasswordCheckEmail', true);
+    }
+
+    private function isAbleToCheckEmail(): bool
+    {
+        $sessionKey = 'ResetPasswordCheckEmail';
+        $session = $this->getSession();
+
+        if ($session->get($sessionKey)) {
+            $session->remove($sessionKey);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    private function storeTokenInSession(string $token): void
+    {
+        $this->getSession()->set('ResetPasswordPublicToken', $token);
+    }
+
+    private function getTokenFromSession(): string
+    {
+        return $this->getSession()->get('ResetPasswordPublicToken');
+    }
+
+    private function getSession(): SessionInterface
```

Comment:
```suggestion
    private function getSessionService(): SessionInterface
```

I don't know... it occurred me that this method could - in theory - conflict with a method they already have in their controller (with the same name). So... maybe `getSessionService()` is a bit more unique???

Reply:
done.. 

Could always suffix all the trait methods with `forReset` or something along those lines to avoid any naming conflicts...